parser-typechecker/src/Unison/Builtin/Decls.hs

```diff
@@ -285,10 +287,27 @@ builtinEffectDecls = []
 
 pattern UnitRef <- (unUnitRef -> True)
 pattern PairRef <- (unPairRef -> True)
+pattern EitherRef <- ((==) eitherRef -> True)
 pattern OptionalRef <- (unOptionalRef -> True)
 pattern TupleType' ts <- (unTupleType -> Just ts)
 pattern TupleTerm' xs <- (unTupleTerm -> Just xs)
 pattern TuplePattern ps <- (unTuplePattern -> Just ps)
+pattern EitherLeft' tm <- (unLeftTerm -> Just tm)
+pattern EitherRight' tm <- (unRightTerm -> Just tm)
+pattern EitherLeftId <- ((==) eitherLeftId -> True)
+pattern EitherRightId <- ((==) eitherRightId -> True)
+
+unLeftTerm, unRightTerm
+  :: Term.Term2 vt at ap v a
+  -> Maybe (Term.Term2 vt at ap v a)
+unRightTerm t = case t of
+  Term.App' (Term.Constructor' EitherRef EitherRightId) tm ->
+    Just tm
+  _                           -> Nothing
+unLeftTerm t = case t of
+  Term.App' (Term.Constructor' EitherRef EitherLeftId) tm ->
+    Just tm
+  _                           -> Nothing
```

Comment:
Optional, if you want to format some of these with ormolu 😬   In vscode it's ⌘K ⌘F to "Format selection" (as opposed to whole file).